src/adapter/src/catalog.rs

```diff
@@ -2012,6 +2099,207 @@ impl<S: Append> Catalog<S> {
+
+        let mut object_queue: VecDeque<_> = migrated_ids.iter().map(|(id, _)| (*id)).collect();
+        let mut visited_set: HashSet<_> = migrated_ids.iter().map(|(id, _)| (*id)).collect();
+        let mut ancestor_ids = HashMap::new();
+
+        let id_fingerprint_map: HashMap<GlobalId, u64> = migrated_ids.into_iter().collect();
+
+        while !object_queue.is_empty() {
+            let id = object_queue.pop_front().unwrap();
+            let entry = self.get_entry(&id);
+
+            let new_id = match id {
+                GlobalId::System(_) => self
+                    .storage()
+                    .await
+                    .allocate_system_ids(1)
+                    .await?
+                    .into_element(),
+                GlobalId::User(_) => self.storage().await.allocate_user_id().await?,
+                _ => unreachable!("can't migrate id: {id}"),
+            };
+
+            // Generate value to update fingerprint and global ID persisted mapping.
+            if let Some(fingerprint) = id_fingerprint_map.get(&id) {
+                let schema_name = self
+                    .get_schema(
+                        &entry.name.qualifiers.database_spec,
+                        &entry.name.qualifiers.schema_spec,
+                        entry.conn_id().unwrap_or(SYSTEM_CONN_ID),
+                    )
+                    .name
+                    .schema
+                    .as_str();
+                migration_metadata.migrated_system_table_mappings.insert(
+                    id,
+                    SystemObjectMapping {
+                        schema_name: schema_name.to_string(),
+                        object_name: entry.name.item.clone(),
+                        id: new_id,
+                        fingerprint: *fingerprint,
+                    },
+                );
+            }
+
+            // Push drop commands.
+            match entry.item() {
+                CatalogItem::Table(_) | CatalogItem::Source(_) => {
+                    migration_metadata.previous_source_ids.push(id)
+                }
+                CatalogItem::Sink(sink) => migration_metadata
+                    .previous_sink_ids
+                    .entry(sink.compute_instance)
+                    .or_default()
+                    .push(id),
+                CatalogItem::Index(index) => migration_metadata
+                    .previous_index_ids
+                    .entry(index.compute_instance)
+                    .or_default()
+                    .push(id),
+                CatalogItem::MaterializedView(mview) => migration_metadata
+                    .previous_materialized_view_ids
+                    .entry(mview.compute_instance)
+                    .or_default()
+                    .push(id),
+                // TODO(jkosh44) Implement log migration
+                CatalogItem::Log(_) => {
+                    panic!("Log migration is unimplemented")
```

Comment:
Is it safe to merge with this? Have we not had any of these yet?

Reply:
The other option is to silently ignore them, which is what we currently do. There seems to be ongoing changes to these which is why I left them out for now. I don't think we've had any log migrations since platform, but if we want to add some then we'll have to implement this.